### PR TITLE
Fix more show

### PR DESCRIPTION
**Pull request by tpoterba**

### CODE REVIEW

**Comment by tpoterba** [comment]
I found another problem while debugging, which is that the select_x methods weren't checking for field uniqueness. I added better checks.

File: python/hail/matrixtable.py [hunk 1]
```diff
@@ -842,18 +841,24 @@ def select_globals(self, *exprs, **named_exprs):
...
+        ids = []
         for e in exprs:
             all_exprs.append(e)
             analyze('MatrixTable.select_globals', e, self._global_indices)
             if e._ast.search(lambda ast: not isinstance(ast, Reference) and not isinstance(ast, Select)):
                 raise ExpressionException("method 'select_globals' expects keyword arguments for complex expressions")
             strs.append(
-                '{}: {}'.format(e._ast.selection if isinstance(e._ast, Select) else e._ast.name, e._ast.to_hql()))
+                '{}: {}'.format(e._ast.name if isinstance(e._ast, Select) else e._ast.name, e._ast.to_hql()))
+            ids.append(e._ast.expand()[0].name)
```
Comment (cseed) [inline python/hail/matrixtable.py]: Don't you just want `e._ast.name` here?  Namely, the name you're putting into strs in the line above?

Also, don't you need to escape the name in the line above?

Comment (tpoterba) [reply python/hail/matrixtable.py]: yeah, that sounds right. I didn't even see this code.

File: python/hail/tests.py [hunk 2]
```diff
@@ -338,6 +338,10 @@ class MatrixTests(unittest.TestCase):
     def get_vds(self, min_partitions=None):
         return hl.import_vcf(test_file("sample.vcf"), min_partitions=min_partitions)
 
+    def test_key_show(self):
```
Comment (cseed) [inline python/hail/tests.py]: Maybe test_row_key_field_show_runs is more precise?

**Comment by tpoterba** [comment]
ahem

**Pull request merged**